commercetools/provider.go

```diff
@@ -55,34 +56,51 @@ func Provider() terraform.ResourceProvider {
 				DefaultFunc: schema.EnvDefaultFunc("CTP_AUTH_URL", nil),
 				Description: "The authentication URL of the commercetools platform. https://docs.commercetools.com/http-api-authorization",
 			},
+			"mc_api_url": {
+				Type:        schema.TypeString,
+				Required:    true,
+				DefaultFunc: schema.EnvDefaultFunc("CTP_MC_API_URL", nil),
+				Description: "The API URL of the Merchant Center. https://docs.commercetools.com/custom-applications/main-concepts/api-gateway#hostnames",
+			},
 		},
 		ResourcesMap: map[string]*schema.Resource{
 			"commercetools_api_client":         resourceAPIClient(),
 			"commercetools_api_extension":      resourceAPIExtension(),
-			"commercetools_subscription":       resourceSubscription(),
-			"commercetools_project_settings":   resourceProjectSettings(),
-			"commercetools_type":               resourceType(),
 			"commercetools_channel":            resourceChannel(),
+			"commercetools_custom_application": resourceCustomApplication(),
 			"commercetools_product_type":       resourceProductType(),
+			"commercetools_project_settings":   resourceProjectSettings(),
 			"commercetools_shipping_method":    resourceShippingMethod(),
+			"commercetools_shipping_zone":      resourceShippingZone(),
 			"commercetools_shipping_zone_rate": resourceShippingZoneRate(),
+			"commercetools_state":              resourceState(),
 			"commercetools_store":              resourceStore(),
+			"commercetools_subscription":       resourceSubscription(),
 			"commercetools_tax_category":       resourceTaxCategory(),
 			"commercetools_tax_category_rate":  resourceTaxCategoryRate(),
-			"commercetools_shipping_zone":      resourceShippingZone(),
-			"commercetools_state":              resourceState(),
+			"commercetools_type":               resourceType(),
 		},
 		ConfigureFunc: providerConfigure,
 	}
 }
 
+// TerraformContext holds the HTTP and GraphQL clients to be used by the Terraform resources.
+// We recommend to use the utility functions `getClient` and `getGraphQLClient`
+// to get the necessary client object.
+type TerraformContext struct {
```

Comment:
This struct is used to hold different values that are used within the resources. Besides the HTTP client, I added the GraphQLClient and the ProjectKey.